pkg/fileio/file_io_test.go

```diff
@@ -74,3 +76,76 @@ func TestCopyFile(t *testing.T) {
 		})
 	}
 }
+
+func TestCopyFileN(t *testing.T) {
+	const (
+		destDirPrefix  = "eib-copy-file-n-test-"
+		srcFilePrefix  = "eib-copy-file-n-test-file-"
+		srcFileContent = "CopyFileN test"
+	)
+
+	tmpDir, err := os.MkdirTemp("", destDirPrefix)
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	tmpFile, err := os.CreateTemp(tmpDir, srcFilePrefix)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, tmpFile.Close())
+		require.NoError(t, os.Remove(tmpFile.Name()))
```

Comment:
Nit: We don't need a file in the first place since `CopyFileN()` accepts an `io.Reader`. This can be simplified by passing `bytes.NewBufferString(srcFileContent)` to each test case and just asserting the string literal.